tools/resourcedocsgen/cmd/metadata_test.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2024, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/hexops/autogold/v2"
+	"github.com/stretchr/testify/require"
+)
+
+func TestMetadataBasic(t *testing.T) {
+	// t.Parallel() //nolint:paralleltest PackageMetadataCmd relies on global state.
+	cmd := PackageMetadataCmd()
+	metadataDir := t.TempDir()
+	pacakgeDocsDir := t.TempDir()
+	cmd.SetArgs([]string{
+		"--repoSlug", "pulumi/pulumi-random",
+		"--schemaFile", "provider/cmd/pulumi-resource-random/schema.json",
+		"--version", "v4.16.7",
+		"--metadataDir", metadataDir,
+		"--packageDocsDir", pacakgeDocsDir,
+	})
+	require.NoError(t, cmd.Execute())
+	t.Run("random.yaml", func(t *testing.T) {
+		t.Parallel()
+		autogold.ExpectFile(t, readFile(t, filepath.Join(metadataDir, "random.yaml")))
+	})
+	t.Run("_index.md", func(t *testing.T) {
+		t.Parallel()
+		autogold.ExpectFile(t, readFile(t, filepath.Join(pacakgeDocsDir, "_index.md")))
```

Comment:
There are some tricks that sometimes make the golden files have fewer escapes:

```
Use your own formatting (JSON, etc.): Make your got value of type autogold.Raw("foobar"), and it will be used as-is for .golden files (not allowed with inline tests.)
```